### DISCUSSION TITLE
In editor: Products appear on sites they are not enabled for

### DISCUSSION THREAD

**Discussion by nitech** [craftcms/commerce#2466] [Ideas]
**Description**
In the editor, products show up on other sites. 

**To Reproduce**
I have four sites.
I've left Product URI Format blank on the sites I don't want the products to appear: 

![image](https://user-images.githubusercontent.com/1279436/116970508-6ac22380-acb8-11eb-8a5b-cc9443c2cdde.png)

**Expected behavior**
I expect the dropdown on Commerce\Products to only contain one site - namely the site that has anything in _Product URI Format_. 

However, as you see, all sites show up: 

![image](https://user-images.githubusercontent.com/1279436/116970379-2df62c80-acb8-11eb-93d8-cabe68d7af2b.png)

This happens only if the current user has access to these specific product types. 
It becomes specifically confusing when the _site selector dropdown box_ selects one of the other sites (that should not have products) because I just edited an entry on one of those sites. 

**Additional info**
- Craft version: Craft Pro 3.6.11.2
- Commerce version: 3.2.17.4
- PHP version: 7.3.3
- Database driver & version: MySQL 5.7.25
- Other Plugins & versions: 

| Plugin | Version | 
| --- | --- |
| Amazon S3 | 1.2.11  | 
| Blitz | 3.8.0  | 
| Bulk Edit | 2.0.5  | 
| Category Groups Field | 1.2.2  | 
| Child Me! | 1.2.0  | 
| CKEditor | 1.0.0-beta.3  | 
| Cloner | 1.2.2  | 
| Control Panel CSS | 2.3.0  | 
| Cookies | 1.1.13  | 
| CP Field Inspect | 1.2.4  | 
| Craft Commerce | 3.2.17.4  | 
| Cryptographer | 1.0.3  | 
| Embedded Assets | 2.5.3  | 
| Enupal Backup | 1.5.0  | 
| Feed Me | 4.3.6  | 
| Formie | 1.3.27  | 
| Imager X | v3.4.0  | 
| Kint | 1.0.0  | 
| Mailchimp Subscribe | 3.0.3  | 
| Navigation | 1.4.15  | 
| Neo | 2.9.5  | 
| NSM Fields | 1.0.2  | 
| Reasons | 2.2.4  | 
| Redactor | 2.8.5  | 
| Retour | 3.1.52  | 
| Scout | 2.4.0  | 
| Section + Product Type | 1.0.5  | 
| Section Field | 1.1.0  | 
| SendGrid | 1.2.2  | 
| SEOmatic | 3.3.39  | 
| Super Table | 2.6.7  | 
| Tags | 1.0.7  | 
| Two-Factor Authentication | 2.9.0  | 
| Typed link field | 1.0.25  | 
| Units | 1.0.4  | 
| Webhooks | 2.3.3  |

**Comment by pdaleramirez** [comment]
@nitech I don't think not displaying product types without a URL in the site dropdown is ideal because there are still other use case where you want to query product with it's translated site on other template pages.

**Comment by nitech** [comment]
@pdaleramirez It is confusing to the user, and it doesn't match the section's logic. 

Take this example: This section is enabled for all but the first site. 
![image](https://user-images.githubusercontent.com/1279436/117413425-3fd50b00-af16-11eb-825e-285ac3db41da.png)

If I look at the Entries menu item in the control panel, I simply won't find this particular section type on the first site. 

If I however look at the Products of the Commerce menu, I will see all of the products on all of the sites- even if I specifically left the Product URI blank "if products don't have URLs"

![image](https://user-images.githubusercontent.com/1279436/117413792-aeb26400-af16-11eb-91a1-3fd7bf5ec298.png)

The most confusing part here is that the products on the sites where products are not enabled, keep the first name the product was given. So if you create a new product named "Saw" and then later rename it to "Japanese Quality Hand Saw", the other languages will still show the product as "Saw".

Have a look at this visual example. I created a product template that the users use as a basis for each new product. On the sites where products are not enabled, this is what I see: 

![image](https://user-images.githubusercontent.com/1279436/117414288-4021d600-af17-11eb-98d0-37b80df7b831.png)

While on the site where the products are enabled, each product have their respective names.

**Comment by lukeholder** [comment]
Right now, products have the same feature-set as categories, they don't act like entries/sections at the moment.

In that you can't disable a product type on a per site basis, only per product basis.

Leaving the URL format blank doesn't disable the product type for that site, it just doesn't create a URL for that product in that site.

We are working on this, as we work on multi-store/multi-site for a future Commerce release.

Reopening, as it is a feature we need to support.

**Comment by nitech** [comment]
Thanks. Looking forward to 4.0
